graph-partitioning/README.rst

```diff
@@ -0,0 +1,82 @@
+
+=== === ==========
+x_i x_j edge (i,j)
+--- --- ----------
+0   0   0
+0   1   1 
+1   0   1
+1   1   0
+=== === ==========
+
+From this table, we see that we can use the expression x_i+x_j-2x_ix_j to calculate the edge column in our table.  Now for our entire graph, our objective function can be written as shown below, where the sum is over all edges in the graph.
+
+.. image:: readme_imgs/QUBO.png
+
+Next we need to consider our constraint:  Subset 0 and Subset 1 must have the same sizes.  We can measure the size of Subset 1 by summing up our binary variables.  To ensure the two subsets have the same size, we enforce a constraint that Subset 1 has size equal to half of all nodes in the graph.  We first consider how to represent this constraint mathematically using our chosen binary variables, and use the following equality to represent our constraint.
+
+.. image:: readme_imgs/constraint_1.png
+
+For a QUBO, we need our constraints to be represented by mathematical expressions that are satisfied at the minimum value.  For this constraint, we can use the following expression that has a minimum value of 0 that occurs when Subset 1 has size exactly `|V|`/2.
+
+.. image:: readme_imgs/constraint_2.png
+
+To simplify this expression and determine the coefficients for our QUBO equation, we first multiply out the expression.
+
+.. image:: readme_imgs/constraint_3.png
+
+Next we can simplify this expression down to linear and quadratic terms for our QUBO.  Recall that for binary variables we can replace any squared term with a linear term (since 0^2=0 and 1^2=1), and we can remove any constant terms in our QUBO.  This results in the following final expression for our constraint.
+
+.. image:: readme_imgs/constraint_4.png
+
+To combine our objective and constraints into a single QUBO expression, we simply add together the objective function and our constraint (multiplied by gamma, the Lagrange parameter).  
+
+.. image:: readme_imgs/final_QUBO.png
+   :align: center
+   :height: 100
+
+In the code, we create the Q matrix for this QUBO as a dictionary iteratively, looping over the edges and nodes in our graph just as we see in the summation of our QUBO expression.
+
+This demo generates an Erdos-Renyi random graph using the ``networkx`` package for our problem instance [1]_.  The graph is constructed with 40 nodes, and each possible edge between two nodes appears with probability p=0.20.
+
+There are three parameters to be set by the user in this code:  chain strength, number of reads, and gamma.  Since this is a relatively large problem, we set a large number of reads (shown on line 23 with ``num_reads = 1000``).  
+
```

Comment:
Is 1000 reads considered large? That's usually my starting point

Reply:
The default value for num_reads is 10, so 1000 is larger than the default :)